server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/RootNodeUpdateCopyNodesByPositionInTree.java

```diff
@@ -38,22 +38,26 @@ public void accept(RootNode node, ProcessingContext ctx) {
   private void updateCopyNodes(RootNode node) {
     List<Node> nodes =
         node.getChildren().stream().filter(Node.hasType(NodeType.COPY)).collect(toList());
-    nodes.forEach(node::removeChild);
-    nodes.forEach(
-        it -> {
-          Node parentNode =
-              RangeUtils.findNodeByPosition(
-                      node, it.getLocality().getUri(), it.getLocality().getRange().getStart())
+    node.getChildren().removeAll(nodes);
+    for (Node it : nodes) {
+      Node parentNode = RangeUtils.findNodeByPosition(node,
+                          it.getLocality().getUri(), it.getLocality().getRange().getStart())
                   .orElse(node);
-            parentNode.addChildAt(getNodeInsertionIndex(parentNode.getChildren(), it), it);
-        });
-
-    List<CopyNode> copyNodes = node.getDepthFirstStream()
-        .filter(Node.hasType(NodeType.COPY))
-        .map(CopyNode.class::cast)
-        .collect(toList());
+      parentNode.addChildAt(getNodeInsertionIndex(parentNode.getChildren(), it), it);
+    }
 
-    copyNodes.forEach(c -> registerCopyUsage(copyNodes, c.getNameLocation(), c.getUri()));
+    List<Node> copyNodes = node.getDepthFirstList(n ->
+            n.getNodeType() == NodeType.COPY && ((CopyNode) n).getUri() != null);
+    for (Node c : copyNodes) {
+      for (Node n : copyNodes) {
+        CopyNode cn1 = (CopyNode) c;
+        CopyNode cn2 = (CopyNode) n;
+        if (!Objects.equals(cn2.getNameLocation(), cn1.getNameLocation())
+                && Objects.equals(cn2.getUri(), cn1.getUri())) {
```

Comment:
This has a different behavior (not necessarily a bug). The new version executes the block when getUri both return null.

Reply:
```((CopyNode) n).getUri() != null);``` should cover this case